src/services/mongoose/wallets.ts

```diff
@@ -102,13 +103,28 @@ export const WalletsRepository = (): IWalletsRepository => {
       return new UnknownRepositoryError(err)
     }
   }
+  const listAllWalletIds = async (
+    walletCurrency: WalletCurrency,
+  ): Promise<WalletId[] | RepositoryError> => {
+    try {
+      const result: WalletRecord[] = await Wallet.find({ currency: walletCurrency })
+      if (!result) {
+        return new CouldNotListWalletIdError()
+      }
+      const walletIds = result.map(({ id }) => id as WalletId)
```

Comment:
`id` should already be WalletId. no cast should be needed

Reply:
I think without casting `id` will be `any` type

Reply:
Casting here is a bug - I think you want
```
 const walletIds = result.map(({ id }) => fromObjectId<WalletId>(id)
```
see src/services/mongoose/utils.ts

Reply:
> I think without casting `id` will be `any` type

don't think it should be any: https://github.com/GaloyMoney/galoy/blob/7f17f41920d5d6fc2b05bc90526f9a6d78b43481/src/services/mongoose/wallets.ts#L116

Reply:
oh ok you're not return the whole `Wallet` object but just `WalletId`

I would return all `Wallet` object, and do the filtering in the domain, so that the function is more generic at this level @thevaibhav-dixit 

thoughts @bodymindarts ?

Reply:
> I would return all Wallet object, and do the filtering in the domain, so that the function is more generic at this level @thevaibhav-dixit

Yes agree - `listByWalletCurrency(currency: WalletCurrency): AsyncGenerator<Wallet | RepositoryError>` would probably be the signature consistent with other similar use cases.

Reply:
Or perhaps `listByWalletCurrency(currency: WalletCurrency): Wallet[] | RepositoryError` - In some places we use `AsyncGenerator` which is preferred but I'm not sure if we use it in combination with `mongoose` or in other situations.